src/doc/rustc/src/platform-support/unknown-uefi.md

```diff
@@ -230,3 +230,71 @@ pub extern "C" fn main(_h: efi::Handle, st: *mut efi::SystemTable) -> efi::Statu
     efi::Status::SUCCESS
 }
 ```
+
+## Rust std for UEFI
+This section contains information on how to use std on UEFI.
+
+### Build std
+The building std part is pretty much the same as the official [docs](https://rustc-dev-guide.rust-lang.org/getting-started.html).
+The linker that should be used is `rust-lld`. Here is a sample `config.toml`:
+```toml
+[llvm]
+download-ci-llvm = false
+[rust]
+lld = true
+[target.x86_64-unknown-uefi]
+linker = "rust-lld"
+```
+Then just build using `x.py`:
+```sh
+./x.py build --target x86_64-unknown-uefi
+```
+
+### Std Requirements
+The current std has a few basic requirements to function:
+1. Memory Allocation Services (`EFI_BOOT_SERVICES.AllocatePool()` and
+   `EFI_BOOT_SERVICES.FreePool()`) are available.
+If the above requirement is satisfied, the Rust code will reach `main`.
+Now we will discuss what the different modules of std use in UEFI.
+
+### Implemented features
+#### alloc
+- Implemented using `EFI_BOOT_SERVICES.AllocatePool()` and `EFI_BOOT_SERVICES.FreePool()`.
+- Passes all the tests.
+- Some Quirks:
+  - Currently uses `EfiLoaderData` as the `EFI_ALLOCATE_POOL->PoolType`.
```

Comment:
Isn't this fairly common? We should probably document it, but what makes it a "quirk"?

Reply:
UEFI has many different memory pools with different semantics. For UEFI-Applications this is certainly the optimal choice. However, if you write UEFI Drivers, you want finer control of the memory-pool (e.g., a uefi-runtime driver needs memory to stay around even after UEFI-services were exited, which is not true for `EfiLoaderData`).

This is all very much related to the allocator-wg and their attempt to allow specifying an allocator. I wouldn't call the decision a `quirk`, but it certainly needs to be documented that the default allocator uses the `EfiLoaderData` pool, and hopefully we can provide additional allocators with the new allocator-generics in the future.